File: app/bll/sshServer.go
package bll

import (
	"go-ssh-forward/app/common"
	"go-ssh-forward/app/dal/sshServer"
)

type SshServerBll struct {
	ssDal *sshServer.SshServerDal
}

func NewSshServerBll(ssDal *sshServer.SshServerDal) *SshServerBll {
	return &SshServerBll{
		ssDal: ssDal,
	}
}

func (b *SshServerBll) List(page, number int) Result {
	results, err := b.ssDal.List(page, number)
	if err != nil {
		return Result{
			Result: false,
			Msg:    err.Error(),
			Data:   nil,
		}
	}
	return Result{
		Result: true,
		Msg:    "success",
		Data:   results,
	}
}

func (b *SshServerBll) Add(data sshServer.ConfigSshServerC) Result {
	if data.Pass != "" {
		data.Pass = common.Encode(data.Pass)
	}
	if data.PassPhrase != "" {
		data.PassPhrase = common.Encode(data.PassPhrase)
	}
	err := b.ssDal.Insert(data)
	return NilDataErr(err)
}

func (b *SshServerBll) Delete(id int) Result {
	err := b.ssDal.Delete(id)
	return NilDataErr(err)
}

func (b *SshServerBll) Update(id int, data sshServer.ConfigSshServerU) Result {
	updates := map[string]interface{}{
		"name":     data.Name,
		"host":     data.Host,
		"user":     data.User,
		"pass":     data.Pass,
		"key_path": data.KeyPath,
		"sort":     data.Sort,
	}

	if data.Pass != "" {
		updates["pass"] = common.Encode(data.Pass)
	}
	if data.ClearPass {
		updates["pass"] = ""
	}

	if data.PassPhrase != "" {
		updates["pass_phrase"] = common.Encode(data.PassPhrase)
	}
	if data.ClearPassPhrase {
		updates["pass_phrase"] = ""
	}
	err := b.ssDal.Update(id, updates)
	return NilDataErr(err)
}

func (b *SshServerBll) GetInfoById(id int) Result {
	data, err := b.ssDal.GetInfoById(id)
	return DataErr(data, err)
}

File: frontend/src/app/pinia/sshServer.ts
import { defineStore } from 'pinia';
import { List } from '@wails/go/bll/SshServerBll';

export type CuSshServerData = {
    name: string;
    host: string;
    user: string;
    pass: string;
    key_path: string;
    pass_phrase: string;
    sort: number;
}
export type SshServerData = CuSshServerData & {
    id: number;
}
export type SshServerEditData = CuSshServerData & {
    id: number;
    pass_new: string;
    pass_phrase_new: string;
    clear_pass: boolean;
    clear_pass_phrase: boolean;
}


type ForwarStore = {
    page: number;
    pageSize: number;
    count: number;
    list: SshServerData[];
    serverList: ElSelectType[];
    fieldName: FieldNameMap;
}

export const useSshServerStore = defineStore('sshServer', {
    state: (): ForwarStore => {
        return {
            page: 1,
            pageSize: 1000,
            count: 0,
            list: [],
            serverList: [],
            fieldName: {
                id: 'ID',
                name: '名称',
                host: '主机',
                user: '用户',
                pass: '密码',
                key_path: '密钥路径',
                pass_phrase: '密钥密码',
                sort: '排序',
            }
        };
    },
    actions: {
        async getList() {
            let result = await List(this.page, this.pageSize);
            if (result.result) {
                let list = result.data as SshServerData[];
                let serverList = list.map((item) => {
                    return {
                        label: item.name,
                        value: item.id,
                    };
                });
                this.serverList = serverList;
                this.list = list;
            }

        },
    }
});